src/modules/charge-versions/controllers/pre-handlers.js

```diff
@@ -46,5 +47,17 @@ const mapInternalCallingUser = (request, h) => {
   return mapOrThrowBoom('user', internalCallingUser, userMapper);
 };
 
+const loadChargeVersionWorkflow = async request => {
+  const { chargeVersionWorkflowId } = request.params;
+  const chargeVersionWorkflow = await chargeVersionWorkflowService.getById(chargeVersionWorkflowId);
+
+  if (!chargeVersionWorkflow) {
+    throw Boom.notFound(`No charge version workflow found with id: ${chargeVersionWorkflowId}`);
```

Comment:
I think you can return rather than throw a Boom error from a hapi pre-handler - this should halt the chain of processing and return the expected status code etc.